lib/docomo_text_to_speech.ex

```diff
@@ -4,65 +4,104 @@ defmodule DocomoTextToSpeech do
        0, 1, 0, 1, 0, 34, 86, 0, 0, 68, 172, 0, 0, 2, 0, 16, 0, 0, 0, 100, 97, 116,
        97, 246, 198, 0, 0, 0, 0, ...>>}
 
+  Without valid API key.
+
+      iex> DocomoTextToSpeech.run "エリクサー"
+      {:error, "(401) POLSLA009: APIKEY is null or empty in the request"}
+
   """
-  def run(
-        text,
-        speaker_id \\ 1,
-        style_id \\ 1,
-        speech_rate \\ 1.0,
-        power_rate \\ 1.0,
-        voice_type \\ 1.0,
-        audio_file_format \\ 2
-      ) do
-    DocomoTextToSpeech.Api.post(
-      text,
-      speaker_id,
-      style_id,
-      speech_rate,
-      power_rate,
-      voice_type,
-      audio_file_format
-    )
-    |> _format_current()
+  @spec run(String.t(), [conversion_option()]) :: {:ok | :error, any()}
+  def run(text, opts \\ []) do
+    params =
+      @default_opts
+      |> Keyword.merge(opts)
+      |> Keyword.put_new(:text_data, text)
+      |> Enum.into(%{}, fn {k, v} ->
+        {k |> to_string() |> Macro.camelize(), v}
+      end)
+
+    "/textToSpeech"
+    |> post(params)
+    |> handle_response()
   end
 
   @doc """
-  Converts text to speech
+  Returns the audio file directly or raise exception if any errors.
 
   ## Examples
 
-      iex> DocomoTextToSpeech.run "エリクサー"
+  Successfull convertion.
+
+      iex> DocomoTextToSpeech.run! "エリクサー"
       <<82, 73, 70, 70, 28, 199, 0, 0, 87, 65, 86, 69, 102, 109, 116, 32, 18, 0, 0,
         0, 1, 0, 1, 0, 34, 86, 0, 0, 68, 172, 0, 0, 2, 0, 16, 0, 0, 0, 100, 97, 116,
         97, 246, 198, 0, 0, 0, 0, ...>>
 
+  Failed conversion.
+
+      iex> DocomoTextToSpeech.run! "エリクサー"
+      ** (RuntimeError) (401) POLSLA009: APIKEY is null or empty in the request
+        (docomo_text_to_speech) lib/docomo_text_to_speech.ex:80: DocomoTextToSpeech.run!/7
+
   """
-  def run!(
-        text,
-        speaker_id \\ 1,
-        style_id \\ 1,
-        speech_rate \\ 1.0,
-        power_rate \\ 1.0,
-        voice_type \\ 1.0,
-        audio_file_format \\ 2
-      ) do
-    {:ok, content} =
-      run(text, speaker_id, style_id, speech_rate, power_rate, voice_type, audio_file_format)
-
-    content
+  @spec run!(String.t(), [conversion_option()]) :: {:ok | :error, any()} | no_return()
+  def run!(text, opts \\ []) do
+    case run(text, opts) do
+      {:ok, content} -> content
+      {:error, msg} -> raise msg
+    end
   end
 
-  defp _format_current({:ok, body, _}), do: {:ok, body}
-  defp _format_current(error), do: error
+  defp api_key do
+    System.get_env("DOCOMO_TEXT_TO_SPEECH_API_KEY") ||
+      Application.get_env(:docomo_text_to_speech, :api_key)
+  end
+
+  defp handle_response({:error, _} = error), do: error
+  defp handle_response({:ok, %Tesla.Env{status: 200, body: body}}), do: {:ok, body}
+
+  defp handle_response({:ok, %Tesla.Env{status: status, body: body}}) do
+    %{"requestError" => %{"policyException" => %{"messageId" => msg_id, "text" => msg}}} = body
+    {:error, "(#{status}) #{msg_id}: #{msg}"}
+  end
```

Comment:
```suggestion
  end
  
  defp camelize("speaker_id"), do: "SpeakerID"

  defp camelize("style_id"), do: "StyleID"

  defp camelize(key), do: Macro.camelize(key)
```

Macro.camelize("speaker_id") == "SpeakerId", but api server wants "SpeakerID".
Macro.camelize("style_id") == "SpeakerId", but api server wants "StyleID".